terraform/environments/example/ec2_complete.tf

```diff
@@ -0,0 +1,257 @@
+        }
+      }
+
+      complete-example-2 = { # The second ec2.
+        tags = {
+          server-type = "private"
+          description = "ec2-complete-example-2"
+          monitored   = false
+          os-type     = "Linux"
+          component   = "ndh"
+          environment = "development"
+        }
+        ebs_volumes = {
+          "/dev/sdf" = { size = 20, type = "gp3" }
+        }
+        ami_name  = "amzn2-ami-kernel-5.10-hvm-2.0.20240131.0-x86_64-gp2"
+        ami_owner = "137112412989"
+        subnet_id = data.aws_subnet.private_subnets_b.id
+        availability_zone = "eu-west-2b"
+        instance_type = "t3.micro"
+        user_data = <<EOF
+            #!/bin/bash
+            yum update -y
+            yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
+            systemctl status amazon-ssm-agent
+            EOF
+        route53_records = {
+          create_internal_record = false
+          create_external_record = false
+        }
+      }
+    }
+  }
+
+  # This local provides a list of ingress and egress rules for the ec2 security group.
+  
+  complete_ec2_sg_ingress_rules = {
+    TCP_22 = {
+      from_port = 22
+      to_port = 22
+      protocol = "TCP"
+      cidr_block = data.aws_vpc.shared.cidr_block
+    }
+    TCP_443 = {
+      from_port = 443
+      to_port = 443
+      protocol = "TCP"
+      cidr_block = data.aws_vpc.shared.cidr_block
+    }
+  }
+
+  complete_ec2_sg_egress_rules = {
+    TCP_ALL = {
+      from_port = 1
+      to_port = 65000
+      protocol = "TCP"
+      cidr_block = "0.0.0.0/0"
+    }
+  }
+
+  # create list of common managed policies that can be attached to ec2 instance profiles
+  ec2_complete_common_managed_policies = [
+    aws_iam_policy.ec2_common_policy.arn
+  ]
+
+}
+
+# This item is used to combine emultiple policy documents though for this example only one policy document is created.
+data "aws_iam_policy_document" "ec2_complete_common_combined" {
+  source_policy_documents = [
+    data.aws_iam_policy_document.ec2_complete_policy.json  
+  ]
+}
+
+# This policy document is added as an example. Note that the module does not support access via AWS Session Manager.
+data "aws_iam_policy_document" "ec2_complete_policy" {
+  statement {
+    sid    = "AllowSSMAccess"
+    effect = "Allow"
+    actions = [
+      "ssm:StartSession",
+      "ssm:ResumeSession",
+      "ssm:TerminateSession",
+      "ssmmessages:CreateControlChannel",
+      "ssmmessages:CreateDataChannel",
+      "ssmmessages:OpenControlChannel",
+      "ssmmessages:OpenDataChannel",
+      "ec2messages:AcknowledgeMessage",
+      "ec2:DescribeInstances"
+    ]
+    resources = ["*"] #tfsec:ignore:aws-iam-no-policy-wildcards
+  }
+}
+
+# This is the main call to the module. Note the for_each loop.
+module "ec2_complete_instance" {
+  source = "github.com/ministryofjustice/modernisation-platform-terraform-ec2-instance?ref=v2.4.1"
```

Comment:
should we use the commit hash? 🤔 

"github.com/ministryofjustice/modernisation-platform-terraform-ec2-instance?ref=edc55b4005b7039e5b54ad7805e89a473fe3c3dd" #v2.4.1

Reply:
Yes good point. The original that I based this from didn't include the ref so I will add that. Thanks for the spot @ASTRobinson

Reply:
Hi @ASTRobinson latest commit has the change. Can you re-review pls :)